rosidl_generator_py/rosidl_generator_py/generate_py_impl.py

```diff
@@ -180,12 +178,7 @@ def constant_value_to_py(type_, value):
         if type_.typename == 'bool':
```

Comment:
@dirk-thomas hmm shouldn't it be `'boolean'`? Also, here and everywhere else, why not use `BOOLEAN_TYPE`?

Reply:
For a single typename it doesn't really make a difference. In cases where the constant represents a set of types it makes sense to use the constant in case the set changes later as well as for readability.

Since there are 100+ cases where single typenames are used as string literals in the various generators I would rather not touch it in this PR.

Reply:
The linter plugins can catch misspellings of `BOOLEAN_TYPE`, but not misspellings of `'bool'`. I don't think it needs to be done in this PR, just pointing out there is a benefit.

Reply:
I do see the improvement in readability, but if the sets were to ever change, having these variables won't do much against the other 100+ cases were the raw string is used (unless it's pure extension, of course). 

In any case, above `shouldn't it be 'boolean'` comment still applies, or otherwise I don't follow why `'boolean'` is used everywhere else when checking `BasicType.typename` attributes.

Reply:
> In any case, above shouldn't it be 'boolean' comment still applies

Absolutely, we currently don't have any boolean constants. Adding one make the code fall through to the exception. I will create a follow PR to fix the behavior (since this doesn't aim to change any behavior).

Reply:
> In any case, above shouldn't it be 'boolean' comment still applies

Absolutely, we currently don't have any boolean constants. Adding one make the code fall through to the exception. I created a follow PR to fix the behavior (since this doesn't aim to change any behavior): #46.